maintainers/scripts/nix-generate-from-cpan.nix

```diff
@@ -1,25 +1,16 @@
-{ stdenv, makeWrapper, perl, perlPackages }:
-
-stdenv.mkDerivation {
-  name = "nix-generate-from-cpan-3";
+{ wrapCommand, lib, perl, perlPackages }:
 
+wrapCommand "nix-generate-from-cpan" {
+  version = "3";
   buildInputs = with perlPackages; [
-    makeWrapper perl CPANMeta GetoptLongDescriptive CPANPLUS Readonly Log4Perl
+    perl CPANMeta GetoptLongDescriptive CPANPLUS Readonly Log4Perl
   ];
-
-  phases = [ "installPhase" ];
-
-  installPhase =
-    ''
-      mkdir -p $out/bin
-      cp ${./nix-generate-from-cpan.pl} $out/bin/nix-generate-from-cpan
-      patchShebangs $out/bin/nix-generate-from-cpan
```

Comment:
So now patchShebangs is no longer being done? Is that intentional? I'm thinking of the case where people want to run the wrapped program directly for any reason – maybe this should be part of makeCommand?

Reply:
Yeah it may be wanted it some cases. In this case perl is used as the interpreter so the shebang will not be messed with. What's nice about this is it can reuse the original path of the file - so in some cases it can be fixed output.

For this one anyway it has the correct shebang (```#!/usr/bin/env perl``` will work everywhere!) So it seemed reasonable to avoid the patchShebangs.

The problem with patchShebangs is that it doesn't differentiate between "nativeBuildInputs" & "buildInputs" - so you can wind up with references from nativeBuildInputs in your output paths if you're not careful. It needs to be reworked to be compatible with cross compilation & strictDeps.

Reply:
> `#!/usr/bin/env perl` will work everywhere!

There is no `/usr/bin/env` in Nix sandbox. (I have not read the context of this PR so I don't know if this is important or not.)